### PR TITLE
sink(ticdc): limit encoder-concurrency to avoid crash

### LINKED ISSUE #11773: CDC OOM when encoder-concurrency is too large
### What did you do?

create a changefeed with config
```
encoder-concurrency = 4294967296
```

### What did you expect to see?

run normally

### What did you see instead?

OOM

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by wk989898**
### What problem does this PR solve?

Issue Number: close #11773
### What is changed and how it works?
As title. The maximum value is CPU count * 10

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/sink/codec/encoder_group.go [hunk 1]
```diff
@@ -69,6 +70,10 @@ func NewEncoderGroup(
 	if concurrency <= 0 {
 		concurrency = config.DefaultEncoderGroupConcurrency
 	}
+	// limit concurrency to avoid OOM
```
Comment (3AceShowHand) [inline pkg/sink/codec/encoder_group.go:73]: It's not to avoid OOM, but to prevent spawn too much goroutine.

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/codec/encoder_group.go [hunk 2]
```diff
@@ -69,6 +70,10 @@ func NewEncoderGroup(
 	if concurrency <= 0 {
 		concurrency = config.DefaultEncoderGroupConcurrency
 	}
+	// limit concurrency to avoid crash
+	if concurrency >= cpu.GetCPUCount()*10 {
+		concurrency = cpu.GetCPUCount() * 10
```
Comment (asddongmen) [inline pkg/sink/codec/encoder_group.go:75]: Please add a warn log here.

**Comment by wk989898** [comment]
/cherry-pick release-8.1 release-7.5 release-7.1 release-6.5 release-6.1

**Comment by wk989898** [comment]
/cherry-pick release-8.1, release-7.5, release-7.1, release-6.5, release-6.1

**Pull request merged**